### PR TITLE
LSNBLDR-585 "Return to Lessons" now dynamic

### PR DESCRIPTION
**Pull request by master-bob**
The Return to Lessons text (when in an external tool through lessons)
is now dynamically labeled. It grabs the page title being used.

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java [hunk 1]
```diff
@@ -1697,6 +1698,18 @@ public String getCurrentToolTitle(String commonToolId) {
 		return tool.getTitle();
 	}
 
+	public String getActualCurrentPageTitle() {
+		String returnVal = "Lessons";
```
Comment (master-bob) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: This is not internationalized and must be fixed.

**Comment by clhedrick** [comment]
I used a slight modified version of this patch, with the title of the Lessons page, not the site page title

**Comment by master-bob** [comment]
I don't think you are being malicious here, but you need to be providing attribution for my work per the ECLv2 (which this work is licensed under, and all work in Sakai for that matter).

There are a couple of ways you could have done this:

1. made a new local branch, cherry-picked my commits, and then made a new commit with the changes you wanted
2. made a new local branch, committed my contributions using the git --author flag, and then made a new commit with your changes

Your work in #2371 doesn't list me as an author whatsoever and credits you for my work. I do not appreciate this.

**Comment by clhedrick** [comment]
As far as I know, no one uses the person who does the commit as attribution. That’s transient information. Normally attribution is in the code, at the top for a whole file or inline for specific sections of the code.

On Apr 27, 2016, at 4:34:30 PM, Leonardo Canessa <notifications@github.com<mailto:notifications@github.com>> wrote:


I don't think you are being malicious here, but you need to be providing attribution for my work per the ECLv2 (which this work is licensed under, and all work in Sakai for that matter).

There are a couple of ways you could have done this:

  1.  made a new local branch, cherry-picked my commits, and then made a new commit with the changes you wanted
  2.  made a new local branch, committed my contributions using the git --author flag, and then made a new commit with your changes

Your work in #2371<https://github.com/sakaiproject/sakai/pull/2371> doesn't list me as an author whatsoever and credits you for my work. I do not appreciate this.

**Comment by jonespm** [comment]
Just as a personal note, I always specify the author as a git commit parameter if I add in a diff from jira or from any source if it's >90% of the change. I have also been careful when migrating any contrib projects from svn to preserving authorship. It can be hard though to find some authors email addresses, but we've been keeping track of most of them. 

I don't know if everyone does that, but if I didn't make a change and can vouch for it, I'd rather someone else be in the git blame for the work than me when a regression comes up. ;)

**Comment by clhedrick** [comment]
OK, I’ll do that if you want. But I thought it could be misleading to give someone else responsibility for a change that in fact differs from the code that they submitted.

On Apr 27, 2016, at 9:03:12 PM, Code Hugger (Matthew Jones) <notifications@github.com<mailto:notifications@github.com>> wrote:


Just as a personal note, I always specify the author as a git commit parameter if I add in a diff from jira or from any source if it's >90% of the change. I have also been careful when migrating any contrib projects from svn to preserving authorship. It can be hard though to find some authors email addresses, but we've been keeping track of most of them.

I don't know if everyone does that, but if I didn't make a change and can vouch for it, I'd rather someone else be in the git blame for the work than me when a regression comes up. ;)

**Comment by clhedrick** [comment]
If people really want to use author this way I think I’d prefer to do two commits, one from their PR and a second to make any of my changes. I know people prefer one commit per patch, but this seems safer.

On Apr 27, 2016, at 9:03:12 PM, Code Hugger (Matthew Jones) <notifications@github.com<mailto:notifications@github.com>> wrote:


Just as a personal note, I always specify the author as a git commit parameter if I add in a diff from jira or from any source if it's >90% of the change. I have also been careful when migrating any contrib projects from svn to preserving authorship. It can be hard though to find some authors email addresses, but we've been keeping track of most of them.

I don't know if everyone does that, but if I didn't make a change and can vouch for it, I'd rather someone else be in the git blame for the work than me when a regression comes up. ;)

**Comment by clhedrick** [comment]
In this specific case I’m going to commit comments to the sections  Canessa wrote.

On Apr 27, 2016, at 9:03:12 PM, Code Hugger (Matthew Jones) <notifications@github.com<mailto:notifications@github.com>> wrote:


Just as a personal note, I always specify the author as a git commit parameter if I add in a diff from jira or from any source if it's >90% of the change. I have also been careful when migrating any contrib projects from svn to preserving authorship. It can be hard though to find some authors email addresses, but we've been keeping track of most of them.

I don't know if everyone does that, but if I didn't make a change and can vouch for it, I'd rather someone else be in the git blame for the work than me when a regression comes up. ;)

**Comment by jonespm** [comment]
Yeah, I agree with that, if you have significant changes to make committing it as two, one with the contributors commit and then your fixup would be the ideal way to go. It sounds if someone hit that github merge and squash option, it would give the contribution credit to the person that opened the pull request only, so if we were using that we might need to be more selective about that but the group that's hitting that button is small enough that it wouldn't be too much of a problem to avoid it 1% of the time.

**Comment by master-bob** [comment]
To avoid the squash issue, you could separate it into two PRs. First PR could be "TICKET  Description" and then second PR could be "TICKET-2 FIXUP", for example.

I think that attribution is meta information that is best contained within the actual commit author. As such comments should be a last resort for attribution as they frequently get out of date.

**Comment by master-bob** [comment]
This same issue is present in PRs #2010 / #2372 , #2023 / #2327, and #522 / #548.

**Comment by clhedrick** [comment]
If the authors of any of these PRs request it I’ll do what I did with LSNBLDR-585. That’s a non-trivial process, and requires QAing it again, so I’m not going to do it unless they ask.

On Apr 28, 2016, at 2:28 PM, Leonardo Canessa <notifications@github.com<mailto:notifications@github.com>> wrote:


This same issue is present in PRs #2010<https://github.com/sakaiproject/sakai/pull/2010> / #2372<https://github.com/sakaiproject/sakai/pull/2372> , #2023<https://github.com/sakaiproject/sakai/pull/2023> / #2327<https://github.com/sakaiproject/sakai/pull/2327>, and #522<https://github.com/sakaiproject/sakai/pull/522> / #548<https://github.com/sakaiproject/sakai/pull/548>.

**Pull request closed without merging**